.github/workflows/continous-delivery.yml

```diff
@@ -85,9 +84,7 @@ jobs:
       - name: Build source and binary distributions
         shell: bash -l {0}
         run: |
-          grep "^tomli\|^setuptools"  requirements/requirements.in | xargs python -m pip install --upgrade --quiet
-          python -m pip install build wheel
-          python setup.py sdist bdist_wheel
+          make dist
 
       - name: Add version to environment vars
         run: |
```

Comment:
After about a year thinking and trying about the question of closed vs open requirements @nromeoarena and I came to the conclusion that for libraries it is better to use open requirements for CI (while tracking closed ones for reference). On the other hand, for applications/services it is better to consider closed requirements for CI/CD.

The reason behind this conclusion is that libraries need to be as flexible as possible to be used in the most diverse cases. To allow this they should be shipped with open requirements. And thus they should be tested in the same configuration as they are shipped (i.e. with open requirements), under all the allowed python versions.
On the other hand, For what it takes the applications, the main focus is on reproducibility and stability across systems. Thus in that case we should use closed requirements (that guarantee reproducibility up to system configuration).

Reply:
I believe we always had this idea. Final applications needs to be stable (therefore installed using pinned requirements.txt fles), while libraries should have open requirements. In poetry you specify the open requirements in your pyproject file (which is shipped when you package the library with `poetry build`)

You can easily export the pinned `requirements.txt` using `poetry export`. Although, if you use poetry, it should not be needed since `poetry.lock` is exactly that (but is however more poetry friendly). 

More info on the CLI [here](https://python-poetry.org/docs/cli/).